code/modules/crafting/table.dm

```diff
@@ -20,18 +20,18 @@
 			return 0
 	return 1
 
-/obj/structure/table/proc/check_table()
+/obj/structure/table/proc/check_table(datum/table_recipe/R)
 	table_contents = list()
 	for(var/obj/item/I in loc)
 		if(istype(I, /obj/item/stack))
 			var/obj/item/stack/S = I
 			table_contents[I.type] += S.amount
 		else
-			if(istype(I, /obj/item/weapon/reagent_containers))
-				for(var/datum/reagent/R in I.reagents.reagent_list)
-					table_contents[R.type] += R.volume
-
 			table_contents[I.type] += 1
+	for(var/obj/item/weapon/reagent_containers/RC in loc)
+		if((RC.flags & OPENCONTAINER) || (R && (RC.type in R.reqs)))
+			for(var/datum/reagent/A in RC.reagents.reagent_list)
+				table_contents[A.type] += A.volume
```

Comment:
1 - I'm pretty sure I did this because I thought it was faster to avoid the istype(reagent_containers) check.

2 - the original code is taking the reagents in all reagent_containers but we want to exclude reagent_containers that aren't open containers (you can't just extract a reagent from one snack and use it for a new recipe), except when the snack is already going to be used in our recipe. That's why we need the recipe variable here.